src/hooks/useUser.js

```diff
@@ -6,29 +6,39 @@ import { transformUserData } from '@utils/data-utils'
 
 export default function useUser(address) {
   const [user, setUser] = useState(null)
+  const [loading, setLoading] = useState()
   const mounted = useMounted()
 
+  useEffect(() => {
+    if (!address && user) {
+      setUser(null)
+    }
+  }, [address, user])
+
   useEffect(() => {
     if (!address) {
       return
     }
 
     const fetchUser = async () => {
       try {
+        setLoading(true)
         const user = await getUser(
           { network: getNetwork().chainId },
           { id: address.toLowerCase() }
         )
+        setLoading(false)
         if (mounted()) {
           setUser(transformUserData(user))
         }
       } catch (err) {
+        setLoading(false)
```

Comment:
We could update the loading state after the try catch once instead of in each block.